dart/dynamics/Shape.h

```diff
@@ -191,29 +142,19 @@ class Shape : public virtual common::Subject
   /// Instruct this shape to update its data
   virtual void refreshData();
 
-  /// \brief
-  virtual void draw(renderer::RenderInterface* _ri = nullptr,
-                    const Eigen::Vector4d& _color = Eigen::Vector4d::Ones(),
-                    bool _useDefaultColor = true) const = 0;
-
-  /// Pass in true to prevent this shape from being rendered. Pass in false to
-  /// allow it to render again.
-  void setHidden(bool _hide=true);
+  /// Notify that the color (rgba) of this shape has updated
+  virtual void notifyColorUpdate(const Eigen::Vector4d& color);
```

Comment:
That said, it is useful to have vertex color data in the visual meshes so that you can create dynamically generated meshes with colors for visualization. I know @mkoval at least has found the need for this in his research, and had to make do with publishing custom visualization messages to ROS.